lib/compat/wordpress-6.7/class-gutenberg-rest-posts-controller-6-7.php

```diff
@@ -0,0 +1,1127 @@
+							'rendered'  => array(
+								'description' => __( 'HTML excerpt for the post, transformed for display.' ),
+								'type'        => 'string',
+								'context'     => array( 'view', 'edit', 'embed' ),
+								'readonly'    => true,
+							),
+							'protected' => array(
+								'description' => __( 'Whether the excerpt is protected with a password.' ),
+								'type'        => 'boolean',
+								'context'     => array( 'view', 'edit', 'embed' ),
+								'readonly'    => true,
+							),
+						),
+					);
+					break;
+
+				case 'thumbnail':
+					$schema['properties']['featured_media'] = array(
+						'description' => __( 'The ID of the featured media for the post.' ),
+						'type'        => 'integer',
+						'context'     => array( 'view', 'edit', 'embed' ),
+					);
+					break;
+
+				case 'comments':
+					$schema['properties']['comment_status'] = array(
+						'description' => __( 'Whether or not comments are open on the post.' ),
+						'type'        => 'string',
+						'enum'        => array( 'open', 'closed' ),
+						'context'     => array( 'view', 'edit' ),
+					);
+					$schema['properties']['ping_status']    = array(
+						'description' => __( 'Whether or not the post can be pinged.' ),
+						'type'        => 'string',
+						'enum'        => array( 'open', 'closed' ),
+						'context'     => array( 'view', 'edit' ),
+					);
+					break;
+
+				case 'page-attributes':
+					$schema['properties']['menu_order'] = array(
+						'description' => __( 'The order of the post in relation to other posts.' ),
+						'type'        => 'integer',
+						'context'     => array( 'view', 'edit' ),
+					);
+					break;
+
+				case 'post-formats':
+					// Get the native post formats and remove the array keys.
+					$formats = array_values( get_post_format_slugs() );
+
+					$schema['properties']['format'] = array(
+						'description' => __( 'The format for the post.' ),
+						'type'        => 'string',
+						'enum'        => $formats,
+						'context'     => array( 'view', 'edit' ),
+					);
+					break;
+
+				case 'custom-fields':
+					$schema['properties']['meta'] = $this->meta->get_field_schema();
+					break;
+
+			}
+		}
+
+		if ( 'post' === $this->post_type ) {
+			$schema['properties']['sticky'] = array(
+				'description' => __( 'Whether or not the post should be treated as sticky.' ),
+				'type'        => 'boolean',
+				'context'     => array( 'view', 'edit' ),
+			);
+		}
+
+		$schema['properties']['template'] = array(
+			'description' => __( 'The theme file to use to display the post.' ),
+			'type'        => 'string',
+			'context'     => array( 'view', 'edit' ),
+			'arg_options' => array(
+				'validate_callback' => array( $this, 'check_template' ),
+			),
+		);
+
+		$schema['properties']['format'] = array(
+			'description' => __( 'The format for the post.' ),
+			'type'        => 'string',
+			'context'     => array( 'view', 'edit' ),
+			'enum'        => array_values( get_post_format_slugs() ),
+		);
```

Comment:
This is the only change to get_item_schema.

Reply:
The `format` property is only included in the schema when the post type supports it above (see `case 'post-formats`). So why do we need unconditional addition here?

Reply:
I don't have a good enough understanding of this. I thought the` $post_type_attributes` and the `format` needed to be separated.

If this needs to be removed, then I don't think `get_item_schema` needs to be in the class extension at all.
Or are you asking me to make it conditional? (if the post type supports post formats) 
Or are you asking me to move it? Add format to post type attributes?

Reply:
If you enable `post-format` for `page` or any custom post type on the trunk, can you set/update it via the post editor?

* If yes, then this code isn't needed.
* If no, this means that `post-formats` are only served for posts, and allowing it for other post types is a matter of different discussion.